File: golang/171. Excel Sheet Column Number .go
func titleToNumber(ct string) int {
    var total float64 = 0
    length := len(ct)
    for i := length - 1; i >= 0; i-- {
        curr := float64(ct[i] - 'A' + 1)
        multi := float64(length - i - 1)
        if multi > 0 {
            curr *= math.Pow(26, multi)
        }
        total += curr
    }
    return int(total)
}

File: java/171. Excel Sheet Column Number .java
public class Solution
{
    public int titleToNumber(String columnTitle)
    {

        if(columnTitle.length() == 0 || columnTitle.length() > 7){
            return 0;
        }

        int sum = 0;

        for(int i=0; i < columnTitle.length(); i++){

            if(!Character.isUpperCase(columnTitle.charAt(i))){      
                return 0;
            }

            int charVal = columnTitle.charAt(i) - 'A' + 1;           
            int exp = columnTitle.length() - i - 1;                
            sum += Math.pow(26, exp) * charVal;                     
        }

        return sum;
    }
}